### PR TITLE
Fix/expunge historical version exception message

### LINKED ISSUE #6879: Improve expungeHistoricalVersion exception to include context (avoid HAPI-2223: null)
**Describe the bug**  
When running an asynchronous Expunge job on the JPA Server, if a historical version record has already been deleted from the database, the server throws an `InternalErrorException` (HAPI-2223) with a `null` message. This makes it impossible to determine which version or partition caused the failure.

**To Reproduce**  
1. Using the JPA Server, create a resource and delete it so that its historical versions exist.  
2. Manually delete one of the corresponding rows from `HFJ_RESOURCE_HISTORY` (e.g., via SQL).  
3. Trigger an expunge operation on that resource (e.g., `POST /fhir/$expunge`). 
4. **Tail or watch the server log** until the background Expunge job finishes—look for an entry like: 
`ERROR ... Error while Expunging`

5. Observe that the thrown exception is `InternalErrorException (HAPI-2223)` with a `null` message.

**Expected behavior**  
The exception message should clearly identify which historical version failed to expunge, for example: `HAPI-2223: No historical version found for ResourceHistoryTablePk(id=12345, partition=1)`

### PR DESCRIPTION
**Pull request by patrick-werner**
## Summary
Improve exception handling in the asynchronous Expunge job so that missing historical versions throw a clear, contextual error instead of HAPI‑2223: null.

## Changes

In JpaResourceExpungeService.expungeHistoricalVersion(), replace orElseThrow(IllegalArgumentException::new) with a lambda that includes the missing version’s ID and partition in the exception message.

Update PartitionRunner’s error‑handling to extract cause.getMessage() and propagate it, ensuring the logged and thrown InternalErrorException carries the real root‑cause text.

Add a simple unit test to verify that an absent history entry yields a non‑null, descriptive exception message.

Related issue
Closes #6879

### CODE REVIEW

**Comment by robogary** [comment]
Formatting check succeeded!

**Comment by robogary** [comment]
Formatting check succeeded!

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/expunge/JpaResourceExpungeService.java [hunk 1]
```diff
@@ -228,8 +229,12 @@ private void expungeHistoricalVersion(
 			RequestDetails theRequestDetails,
 			ResourceHistoryTablePk theNextVersionId,
 			AtomicInteger theRemainingCount) {
-		ResourceHistoryTable version =
-				myResourceHistoryTableDao.findById(theNextVersionId).orElseThrow(IllegalArgumentException::new);
+		ResourceHistoryTable version = myResourceHistoryTableDao
+				.findById(theNextVersionId)
+				.orElseThrow(() -> new IllegalArgumentException(MessageFormat.format(
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/expunge/JpaResourceExpungeService.java:234]: I guess we should probably include an error code here - We try to include them on all newly thrown exception messages that might end up visible to an end user. (FYI we use [this page](https://github.com/hapifhir/hapi-fhir/wiki/Bumping-Error-Code) to reserve them)

```suggestion
				.orElseThrow(() -> new IllegalArgumentException(Msg.code(2701) + MessageFormat.format(
```

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/expunge/JpaResourceExpungeServiceTest.java [hunk 2]
```diff
@@ -56,4 +61,32 @@ public void testExpungeDoesNotDeleteAllSearchParams() {
...
 	}
+
+
+	@Test
+	public void testExpungeHistoricalVersions_missingEntry_throwsWithContextualMessage() throws NoSuchFieldException, IllegalAccessException {
+		// given a primary key with no existing history
+		ResourceHistoryTablePk pk = new ResourceHistoryTablePk();
+		pk.setPartitionIdValue(42);
+		// set versionId via reflection to avoid null in message
+		Field versionIdField = ResourceHistoryTablePk.class.getDeclaredField("myVersionId");
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/expunge/JpaResourceExpungeServiceTest.java:72]: IMO we could just add a setter to this field so the test has better type safety... No harm in adding it, it's just not there 'cause it wasn't needed before

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/expunge/JpaResourceExpungeServiceTest.java [hunk 3]
```diff
@@ -56,4 +61,32 @@ public void testExpungeDoesNotDeleteAllSearchParams() {
...
+		versionIdField.setAccessible(true);
+		versionIdField.set(pk, 123L);
+
+		when(myResourceHistoryTableDao.findById(pk)).thenReturn(Optional.empty());
+
+		AtomicInteger remaining = new AtomicInteger(1);
+		IllegalArgumentException ex = assertThrows(
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/expunge/JpaResourceExpungeServiceTest.java:79]: Nitpick: This is fine, but @fil512 just pointed out to me how much nicer the assertThatThrownBy syntax can be for this kind of assertion the other day: https://github.com/hapifhir/hapi-fhir/blob/75162c57f02e9043bd4cfc4ef5bfa93c64c310b3/hapi-fhir-base/src/test/java/ca/uhn/fhir/interceptor/executor/InterceptorServiceTest.java#L698-L700

**Review by jamesagnew — APPROVED** [review]

**Comment by robogary** [comment]
Formatting check succeeded!

**Comment by patrick-werner** [comment]
@jamesagnew thanks for your review, i applied all recommended changes.
As i lost my merge rights, can you please merge?
